src/com/palmergames/bukkit/towny/object/TownBlockTypeCache.java

```diff
@@ -0,0 +1,89 @@
+package com.palmergames.bukkit.towny.object;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class TownBlockTypeCache {
+
+	private Map<TownBlockType, Integer> typeCache = new ConcurrentHashMap<>();
+	private Map<TownBlockType, Integer> forSaleCache = new ConcurrentHashMap<>();
+	private Map<TownBlockType, Integer> residentOwnedCache = new ConcurrentHashMap<>();

```

Comment:
Initializing the bottom two maps with an initialCapacity of 0 might reduce memory use a small bit.